youtube_dl/extractor/vimeo.py

```diff
@@ -482,10 +482,15 @@ def _extract_urls(url, webpage):
             r'<embed[^>]+?src=(["\'])(?P<url>(?:https?:)?//(?:www\.)?vimeo\.com/moogaloop\.swf.+?)\1',
             # Look more for non-standard embedded Vimeo player
             r'<video[^>]+src=(["\'])(?P<url>(?:https?:)?//(?:www\.)?vimeo\.com/[0-9]+)\1',
+            # Look for JS embedded Vimeo player
+            r'<video-player[^>]+vimeo-id="(?P<id>\d+)"',
         )
         for embed_re in PLAIN_EMBED_RE:
             for mobj in re.finditer(embed_re, webpage):
-                urls.append(mobj.group('url'))
+                if mobj.group('id'):
+                    urls.append('https://player.vimeo.com/video/' + mobj.group('id'))
+                else:
+                    urls.append(mobj.group('url'))
```

Comment:
Sorry for this Pull Request has long been reviewed. May I suggest adding referrer to the embed link, as suggested below?

```suggestion
                if mobj.group('id'):
                    urls.append(VimeoIE._smuggle_referrer('https://player.vimeo.com/video/' + mobj.group('id'), url))
                else:
                    urls.append(mobj.group('url'))
```

I came across this Pull Request, since I was looking at Issue #24381. Adding vimeo-id tag support would help downloading video from Laracasts.com too. Although, we still need to implement an extra Extractor for playlist.

Hopefully, your Pull Request can be approved.